keras/backend/torch/nn.py

```diff
@@ -112,20 +111,30 @@ def _compute_padding_length(
     input_length, kernel_length, stride, dilation_rate=1
 ):
     """Compute padding length along one dimension."""
-    if (input_length - 1) % stride == 0:
-        total_padding_length = dilation_rate * (kernel_length - 1)
-    else:
-        total_padding_length = (
-            dilation_rate * (kernel_length - 1) - (input_length - 1) % stride
-        )
-    left_padding = int(np.floor(total_padding_length / 2))
-    right_padding = int(np.ceil(total_padding_length / 2))
+    total_padding_length = (
+        dilation_rate * (kernel_length - 1) - (input_length - 1) % stride
+    )
+    left_padding = total_padding_length // 2
+    right_padding = (total_padding_length + 1) // 2
     return (left_padding, right_padding)
 
 
 def _apply_same_padding(
     inputs, kernel_size, strides, operation_type, dilation_rate=1
 ):
+    """Apply same padding to the input tensor.
+
+    This function will evaluate if the padding value is compatible with torch
+    functions. To avoid calling `pad()` as much as possible, which may cause
+    performance or memory issues, when compatible, it does not apply the padding
+    to the tensor, but returns the input tensor and the padding value to pass to
+    the torch functions. If not compatible, it returns the padded tensor and 0
+    as the padding value.
+
+    Returns:
+        tensor: A padded tensor or the inputs.
+        padding: The padding value, ready to pass to the torch functions.
+    """
     spatial_shape = inputs.shape[2:]
     num_spatial_dims = len(spatial_shape)
     padding = ()
```

Comment:
Looks like it needs to be standardized to a `left, right` tuple.

Reply:
fixed. Thanks!